### PR TITLE
Fix peer dependencies

### PR DESCRIPTION
**Pull request by DragonRaider5**
As I was about to switch to peer and dev dependencies, I discovered this'd already been done.

However I noticed the versioning was faulty (`^16.3.0` would only accept `16.3.x`) and the `react` dev dependency also didn't match the requested version of `react-dom`. In addition the selected version didn't support hooks (needed for unit testing them).

Edit: Typo, grammar

### CODE REVIEW

**Comment by longlho** [comment]
`^16.3.0` includes all `16.x` https://semver.npmjs.com/

**Comment by DragonRaider5** [comment]
You're right, I was pretty sure it only matches the same minor 👍 

Rest of the fixes are still valid? I think something along the lines of `>= 16.3.0 < 17.0.0` would still be more verbose, but that's just personal preference.

Do we want to ban future major releases of `react`?

**Comment by longlho** [comment]
hmm yeah the rest looks good. `^16.3.0` also prevents `17.0.0` but I don't mind either way

File: package.json [hunk 1]
```diff
@@ -46,8 +46,8 @@
     "shallow-equal": "^1.1.0"
   },
   "peerDependencies": {
-    "@types/react": "^16.0.0",
-    "react": "^16.3.0"
+    "@types/react": ">= 16.3.0",
```
Comment (longlho) [inline package.json]: yeah this makes us allow `17.0.0` so might be dangerous

File: package.json [hunk 2]
```diff
@@ -60,7 +60,7 @@
     "@types/enzyme": "^3.10.3",
     "@types/jest": "^24.0.17",
     "@types/prop-types": "^15.7.1",
-    "@types/react": "^16.0.0",
+    "@types/react": "^16.9.2",
```
Comment (redonkulus) [inline package.json]: @longlho we want to keep this at ^16.0.0 due to some bug right?

File: package.json (hunk 1 above)
Comment (redonkulus) [reply package.json]: Agreed we should only be using ^16.0.0

File: package.json (hunk 2 above)
Comment (longlho) [reply package.json]: `devDep` is ok I think

**Pull request closed without merging**